### PR TITLE
Peer review for code change to include Azure Linux installations[[install-powershell.sh doesn't work on azure-linux #23944]]

### LINKED ISSUE #23944: install-powershell.sh doesn't work on azure-linux
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

I deployed a virtual machine in azure specifying:

publisher: MicrosoftCBLMariner
offer: azure-linx
sku: azure-linux-gen2

when I ran the install script, it failed saying it was an unsupported os:

`Sorry, your operating system is based on  and is not supported by PowerShell or this installer at this time`

### Expected behavior

```console
pwsh is installed
```

### Actual behavior

```console
pswh installation fails with error:

Sorry, your operating system is based on  and is not supported by PowerShell or this installer at this time
```

### Error details

```console
Sorry, your operating system is based on  and is not supported by PowerShell or this installer at this time
```

### Environment data

```powershell
N/A
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by bosesubham2011**
…on azure-linux 

Fix #23944

# PR Summary

This change will essentially allow PowerShell to be installed on Azure Linux boxes. As it is known that Azure Linux has a mariner base Distribution System, so it was only prudent to add the piece of code to allow for installation on Azure Linux boxes.

## PR Context

This helps resolve - "install-powershell.sh doesn't work on azure-linux #23944" issue
## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
  - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
  - [ ] None
  - **OR**
  - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.5/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
    - [ ] Experimental feature name(s): 
- **User-facing changes**
  - [ ] Not Applicable
  - **OR**
  - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
  - [ ] N/A or can only be tested interactively
  - **OR**
  - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
  - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
  - **OR**
  - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
    - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
      - [ ] Issue filed: 
    - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
      - [ ] Issue filed: 
    - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
      - [ ] Issue filed: 
    - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
      - [ ] Issue filed:

### CODE REVIEW

**Review by dpoole73 — APPROVED** [review]

File: tools/install-powershell.sh [hunk 1]
```diff
@@ -121,6 +121,9 @@ install(){
 			if [[ $osname = *SUSE* ]]; then
 				DistroBasedOn='suse'
 				REV=$(source /etc/os-release; echo $VERSION_ID)
+            elif [ $osname = *"Azure Linux"* ] ; then
```
Comment (dpoole73) [inline tools/install-powershell.sh:124]: shouldn't this be `[[ $osname = *"Azure Linux"* ]]`

Comment (dpoole73) [reply tools/install-powershell.sh:124]: also maybe a better way would be to follow the same pattern above?  eg:
```
   elif [ -f /etc/azurelinux-release ] ; then
                DistroBasedOn='mariner'
                PSUEDONAME=$( (sed s/.*\(// | sed s/\)//) < /etc/azurelinux-release )
                REV=$( (sed s/.*release\ // | sed s/\ .*//) < /etc/azurelinux-release )

Comment (bosesubham2011) [reply tools/install-powershell.sh:124]: I agree this is a better way to solve this issue.

**Review by dpoole73 — APPROVED** [review]

File: tools/install-powershell.sh (hunk 1 above)
Comment (dpoole73) [reply tools/install-powershell.sh]: i tried it and it doesn't work:

Get-PowerShell MASTER Installer Version 1.2.0
Installs PowerShell and Optional The Development Environment
  Original script is at: https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools\install-powershell.psh
Arguments used:

Operating System Details:
  OS: linux
  DIST:
  DistroBasedOn: mariner
  PSUEDONAME: Azure Linux 3.0
AZURELINUX_BUILD_NUMBER=3.0.20240524
  REV: Azure
AZURELINUX_BUILD_NUMBER=3.0.20240524
  KERNEL: 6.6.29.1-3.azl3
  MACH: x86_64
  OSSTR:
Configuring PowerShell Environment for: mariner  Azure
AZURELINUX_BUILD_NUMBER=3.0.20240524
Could not find "installpsh-mariner.sh" next to this script...
Pulling and executing it from "https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools/installpsh-mariner.sh"
found and using curl

*** PowerShell Development Environment Installer 1.2.0 for mariner
***    Original script is at: https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools/installpsh-mariner.sh

*** Arguments used:

*** This installer is only for mariner and you are running , please run "https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools\install-powershell.sh" to see if your distro is supported AND to auto-select the appropriate installer if it is.

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Pull request merged**